peer-assessment-app/app/api/auth/[...nextauth]/route.ts

```diff
@@ -1,55 +1,94 @@
-            // Execute a query to find the user by email
-            const rows = await db.execute('SELECT * FROM users WHERE email = ?', [credentials?.email]);
-            const user = rows[0];
-  
-            if (!user) {
-              return null;
-            }
-  
-            // Compare the provided password with the stored hashed password
-            const passwordCorrect = await compare(credentials?.password || "", user.password);
-  
-            if (passwordCorrect) {
-              return {
-                id: user.id,
-                email: user.email,
-              };
-            }
-          } catch (error) {
-            console.error('Database error:', error);
-          }
-  
-          return null;
-        },
-      }),
+        CredentialsProvider({
+            name: "Credentials",
+            credentials: {
+                email: { label: "Email", type: "text" },
+                password: { label: "Password", type: "password" },
+                userType: { label: "User Type", type: "text" },
+            },
+            async authorize(credentials) {
+                try {
+                    const db = await mysql.createConnection(GetDBSettings());
+
+                    const email = credentials?.email;
+                    const password = credentials?.password;
+                    const userType = credentials?.userType;
+
+                    if (!email || !password || !userType) {
+                        return null;
+                    }
+
+                    let query = '';
+                    let params: any = [];
+                    let idField = '';
+                    let nameField = '';
+                    let emailField = '';
+                    let passwordField = '';
+
+                    if (userType === 'student') {
+                        query = 'SELECT s_id as id, s_name as name, s_email as email, s_password as password FROM students WHERE s_email = ?';
+                        params = [email];
+                        idField = 'id';
+                        nameField = 'name';
+                        emailField = 'email';
+                        passwordField = 'password';
+                    } else if (userType === 'instructor') {
+                        query = 'SELECT i_id as id, i_name as name, i_email as email, i_password as password FROM instructors WHERE i_email = ?';
+                        params = [email];
+                        idField = 'id';
+                        nameField = 'name';
+                        emailField = 'email';
+                        passwordField = 'password';
+                    } else {
+                        // Invalid user type
+                        return null;
+                    }
+
+                    const [rows]: any = await db.execute(query, params);
+
+                    if (rows.length === 0) {
+                        // No user found
+                        return null;
```

Comment:
return null? should you not return a next response instead?

Reply:
The null return is handled by 
           # const result = await signIn('credentials', {
            redirect: false,
            email: data.get('email'),
            password: data.get('password'),
            userType: userType,
        });